### ISSUE TITLE
Error about no available topolog found on openshift. 

### ISSUE THREAD

**Issue by kundeng** [openebs/lvm-localpv#344]
**What steps did you take and what happened:**
installed the latest helm chart, manually created the VG. 

failed to provision volume with StorageClass "openebs-lvmpv": error generating accessibility requirements: no available topology found

**What did you expect to happen:**
provision succeeds

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other [Pastebin](https://pastebin.com/) is fine.)

* `kubectl logs -f openebs-lvm-localpv-controller-7b6d6b4665-fk78q -n openebs -c openebs-lvm-plugin`

* `kubectl logs -f openebs-lvm-localpv-node-[xxxx] -n openebs -c openebs-lvm-plugin`
```
I1031 21:39:28.254386       1 main.go:149] LVM Driver Version :- 1.6.1 - commit :- a3c08bad6ad3ae64581f135e076d20b2ca26636c
I1031 21:39:28.254699       1 main.go:150] DriverName: local.csi.openebs.io Plugin: controller EndPoint: unix:///var/lib/csi/sockets/pluginproxy/csi.sock NodeID:  SetIOLimits: false ContainerRuntime: containerd RIopsPerGB: [] WIopsPerGB: [] RBpsPerGB: [] WBpsPerGB: []
I1031 21:39:28.255033       1 driver.go:49] enabling volume access mode: SINGLE_NODE_WRITER
I1031 21:39:28.268530       1 controller.go:213] waiting for k8s & lvm node informer caches to be synced
I1031 21:39:28.369081       1 controller.go:217] synced k8s & lvm node informer caches
I1031 21:39:28.369115       1 controller.go:219] initializing csi provisioning leak protection controller
I1031 21:39:28.393450       1 leak_protection.go:134] "starting up csi pvc controller"
I1031 21:39:28.394025       1 shared_informer.go:311] Waiting for caches to sync for CSI Provisioner
I1031 21:39:28.395310       1 shared_informer.go:318] Caches are synced for CSI Provisioner
I1031 21:39:28.396617       1 grpc.go:190] Listening for connections on address: &net.UnixAddr{Name:"//var/lib/csi/sockets/pluginproxy/csi.sock", Net:"unix"}
I1031 21:39:28.620969       1 grpc.go:72] GRPC call: /csi.v1.Identity/Probe requests {}
I1031 21:39:28.623770       1 grpc.go:81] GRPC response: {}
I1031 21:39:28.626482       1 grpc.go:72] GRPC call: /csi.v1.Identity/GetPluginInfo requests {}
I1031 21:39:28.626583       1 grpc.go:81] GRPC response: {"name":"local.csi.openebs.io","vendor_version":"1.6.1"}
I1031 21:39:28.660649       1 grpc.go:72] GRPC call: /csi.v1.Identity/GetPluginCapabilities requests {}
I1031 21:39:28.660760       1 grpc.go:81] GRPC response: {"capabilities":[{"Type":{"Service":{"type":1}}},{"Type":{"Service":{"type":2}}}]}
I1031 21:39:28.662986       1 grpc.go:72] GRPC call: /csi.v1.Controller/ControllerGetCapabilities requests {}
I1031 21:39:28.663220       1 grpc.go:81] GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":9}}},{"Type":{"Rpc":{"type":5}}},{"Type":{"Rpc":{"type":4}}}]}
I1031 21:39:28.663795       1 grpc.go:72] GRPC call: /csi.v1.Controller/ControllerGetCapabilities requests {}
I1031 21:39:28.663838       1 grpc.go:81] GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":9}}},{"Type":{"Rpc":{"type":5}}},{"Type":{"Rpc":{"type":4}}}]}
I1031 21:39:28.795749       1 grpc.go:72] GRPC call: /csi.v1.Identity/GetPluginInfo requests {}
I1031 21:39:28.795806       1 grpc.go:81] GRPC response: {"name":"local.csi.openebs.io","vendor_version":"1.6.1"}
I1031 21:39:28.802674       1 grpc.go:72] GRPC call: /csi.v1.Identity/Probe requests {}
I1031 21:39:28.802711       1 grpc.go:81] GRPC response: {}
I1031 21:39:28.804416       1 grpc.go:72] GRPC call: /csi.v1.Controller/ControllerGetCapabilities requests {}
I1031 21:39:28.804454       1 grpc.go:81] GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":9}}},{"Type":{"Rpc":{"type":5}}},{"Type":{"Rpc":{"type":4}}}]}
I1031 21:39:29.259262       1 grpc.go:72] GRPC call: /csi.v1.Identity/Probe requests {}
I1031 21:39:29.259426       1 grpc.go:81] GRPC response: {}
I1031 21:39:29.267784       1 grpc.go:72] GRPC call: /csi.v1.Identity/GetPluginInfo requests {}
I1031 21:39:29.267846       1 grpc.go:81] GRPC response: {"name":"local.csi.openebs.io","vendor_version":"1.6.1"}
I1031 21:39:29.271501       1 grpc.go:72] GRPC call: /csi.v1.Identity/GetPluginCapabilities requests {}
I1031 21:39:29.271552       1 grpc.go:81] GRPC response: {"capabilities":[{"Type":{"Service":{"type":1}}},{"Type":{"Service":{"type":2}}}]}
I1031 21:39:29.273316       1 grpc.go:72] GRPC call: /csi.v1.Controller/ControllerGetCapabilities requests {}
I1031 21:39:29.273355       1 grpc.go:81] GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":9}}},{"Type":{"Rpc":{"type":5}}},{"Type":{"Rpc":{"type":4}}}]}
```

* `kubectl get pods -n openebs`
```
NAME                                              READY   STATUS    RESTARTS      AGE
openebs-localpv-provisioner-6bd66f8598-6d2fb      1/1     Running   3 (51m ago)   63m
openebs-lvm-localpv-controller-776996844c-p6hkj   5/5     Running   0             38m
openebs-zfs-localpv-controller-b8bc9ff46-9dd6g    5/5     Running   0             63m
```

* `kubectl get lvmvol -A -o yaml`
kubectl get lvmvol -A -o yaml
apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""

**Environment:**
- LVM Driver version
- Kubernetes version (use `kubectl version`):
Client Version: v1.30.4
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.4
- Kubernetes installer & version: openshift 4.17
- Cloud provider or hardware configuration:  vsphere VM
- OS (e.g. from `/etc/os-release`): redhat core-os

**Comment by kundeng** [comment]
sh-5.1# pvdisplay 
  --- Physical volume ---
  PV Name               /dev/sdb
  VG Name               lvmvg
  PV Size               200.00 GiB / not usable 4.00 MiB
  Allocatable           yes 
  PE Size               4.00 MiB
  Total PE              51199
  Free PE               51199
  Allocated PE          0
  PV UUID               LIZivF-rcQz-3dPl-heDh-HvMQ-l43O-EgAnNx

**Comment by kundeng** [comment]
The issue is we need: 

oc adm policy add-scc-to-user hostmount-anyuid -z openebs-lvm-node-sa -n openebs


or 
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: openebs-lvm-node-sa-binding
subjects:
  - kind: ServiceAccount
    name: openebs-lvm-node-sa
    namespace: openebs
roleRef:
  kind: ClusterRole
  name: hostmount-anyuid
  apiGroup: rbac.authorization.k8s.io
```

**Comment by dsharma-dc** [comment]
From the log snippet it appears provision request hasn't even come to lvm provisioner. Could you share the storage class definition please?

**Comment by kundeng** [comment]
> From the log snippet it appears provision request hasn't even come to lvm provisioner. Could you share the storage class definition please?

See the solution I provided.  helm chart need to incorporate that.

**Comment by abhilashshetty04** [comment]
@kundeng , Please also share `csi-provisioner` and `csi-controller` container log along with sc output.

 If you could collect logs while you reproduce that would be Good.

**Comment by abhilashshetty04** [comment]
@kundeng  Can you please raise the PR with required changes?
